plugins/signals-react/src/hooks/useSignals.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState } from 'react';
+import { SignalsClient } from './SignalsClient';
+import {
+  discoveryApiRef,
+  identityApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+import { UseSignalsOptions } from './types';
+
+/**
+ * Global signals client for frontend
+ *
+ * @private
+ */
+let globalSignalsClient: SignalsClient | null;
+
+/**
+ * Use signals API to subscribe to events.
+ *
+ * @param pluginId - plugin id
+ * @param onMessage - callback for message
+ * @param options - additional options
+ *
+ * @public
+ */
+export function useSignals(options: UseSignalsOptions) {
+  const [subscriptionKey, setSubscriptionKey] = useState<string | null>(null);
+  const identityApi = useApi(identityApiRef);
+  const discoveryApi = useApi(discoveryApiRef);
+
+  if (!globalSignalsClient) {
+    globalSignalsClient = SignalsClient.create({ identityApi, discoveryApi });
```

Comment:
Let's make the signals client a utility API as well. If this functionality is only accessible via a React hook then other utility APIs and non-react code won't be able to use it. We'll also want to make sure there's only one signals client instance right? Part of the benefit of this plugin is to be able handle all communication through one place.